### DISCUSSION TITLE
중급 프로젝트 최종 팀회고

### DISCUSSION THREAD

**Discussion by sulyoungoh** [codeit-bootcamp-frontend/3-Weekly-Mission#339] [회고]
치열했던 중급 프로젝트 시간을 보내고 벌써 다음주에 발표를 앞두고 있네요 여러분 ☺️

중급 프로젝트 최종 팀 회고를 하고([회고 방식 상세보기](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/discussions/337#discussion-6334449)), 이전처럼 댓글 남겨주세요.
서로 고생 많았다고 다독여주는 시간, 그리고 아쉬웠던 부분은 회고할 수 있는 시간 되었으면 좋겠어요.  

중급 프로젝트 너무 너무 고생많으셨습니다 💜

**Comment by mingzzi96** [comment]
# 7팀

## KEEP
- 소통이 원활하고 참석이 잘 된다.
- 아직 모르것이 많아 기능구현에서 어려운점은 분명히 있지만 팀원 모두 하려고 하는 의지들이 강하다
- 집단지성의 힘은 대단하다. 

## PROBLEM
- 오류 해결 기록을 조금 더 자주 해둘걸 후회했다.
- 어쩌다 해결된 문제들을 그자리에서 복기해보지 않아 왜 해결됐는지 까먹어버렸다
- 디버깅하는 연습이 필요하다.

## TRY
- 진행 상황 자세하게 체크해야겠다
- 나중에 디테일 수정하더라도 뭔가 막히던게 해결됐을때 어떻게 했는지 간단하게라도 적어두자
- 투두 리스트를 해보려고 했던거 중간중간 까먹지 말고 진행해보자.

**Comment by nyaknya** [comment]
# 6팀

## KEEP
- 모두들 책임감을 가지고 맡은 일에 최선을 다한다.
- 모르는 부분이 있으면 서로 도와가며 한다.
- 연락이 빠르게 잘 된다. 
- 각자의 기능 구현에 사용된 코드를 볼 일이 잦아서 상대방 코드의 장단점을 볼 기회가 많아서 좋았다.

## PROBLEAM 
- 시간이 예상보다 촉박해서 기능 구현에 집중하느라 챙기지 못한 것들이 많은 듯 싶다.
- 특히 Tailwind를 처음 사용해서 그런지 디자인적인 측면에서 아쉬운점이 많이 보인다.
- 짜여져 있는 API 내에서 프로젝트를 진행하다보니 API로직 관련으로도 원하는 기능을 구현하지 못하는 경우가 많다는걸 종종 느꼈다.
- 아직은 팀프로젝트에 익숙하지는 않아서 서로의 로직이 자주 연결되는 프로젝트를 하다보니 결과적으로 불필요한 코드들도 있고, 컨벤션이 깔끔하게 지켜지지 않은 부분도 있는거같다.

## TRY 
- 앞으로의 프로젝트에서는 진행 중간중간 다 같이 리팩토링을 거쳐서 코드를 정리하고 다음 기능 개발을 하는 방법을 시도해보고싶다.
- 이번 프로젝트도 종료 후 기록해 놓은 문제점들을 리팩토링 하는 시간을 가져야겠다.

**Comment by DanteSnow** [comment]
# 🔥2팀🔥

## KEEP
팀원 모두가 자신이 원래 맡은 역할이 아닌데도, 솔선수범하여 작업들을 맡아서 일정이 수월하게 진행됐습니다.
적당한 잡담과 농담들 덕분에 프로젝트가 즐거웠습니다. 분위기가 너무 좋았습니다.
이번 프로젝트를 통해서 처음 배워가는 것이 정말 많았습니다. 다른 팀원 분들의 지식을 배워가는 느낌이어서 감사했습니다.
프로젝트 첫 시작부터 끝날 때 까지 서로가 서로를 배려하고 존중하는 분위기가 형성되어 있었습니다.

정말 3주도 안되는 기간이었지만 이번 프로젝트를 하면서 너무 많은 점을 배울 수 있었던 것 같습니다. 기초 프로젝트 떄에 비해 난이도도 올라가며 전에는 사용해 보지 않았던 폴더 구조와 여러 라이브러리를 사용해 어려움이 있었습니다. 하지만 팀원분들께서 먼저 도와주시려고 하시는 모습과 서로의 문제점을 같이 해결해주시려는 모습으로 제가 중급 프로젝트를 완성할 수 있었다고 생각하고 있습니다. 치현님의 경우 정말 팀장의 표본이라고 할 수 있을 정도로, 팀원들을 잘 이끌어 주셨으며 항상 늦은 시간까지 고민하시고 해결하려고 하셨습니다. 치현님이 이번 프로젝트의 높은 완성도에 정말 큰 역할을 해주셨던 것 같습니다. 시열님은 항상 빠른 속도로 작업을 해주셔서 옆에서 정말 자극을 많이 받을 수 있었던 것 같습니다. 그리고 시열님의 코드를 보면서 많이 배웠던 것 같습니다. 윤성님은 항상 새로운 것에 도전하려고 하시며, 문제점이 생기면 끝까지 해결하시려는 모습을 보여주셨습니다. 그리고 팀원들이 놓칠 수 있었던 사소한 부분들까지 확인해 주셔서 프로젝트가 원활하게 진행되었던 것 같습니다. 이번 프로젝트가 개인적으로 어려웠지만 좋은 팀원들을 만나 끝까지 잘 완성할 수 있었던 것 같습니다!

팀원분들께서 맡은 역할을 끝까지 수행하기위해 노력하는 모습이 좋았습니다
말하지 않아도 각자 버그나 부족한 부분을 메꾸려고 하는 모습이 좋았습니다.
팀원분들께서 항상 늦은 시간까지 작업을 하시고, 어려운 부분도 솔선수범 하시는 모습이 귀감이 된것 같습니다

맡은 역할 책임감 있게 해내는 모습이 좋았습니다.
어려운 부분이 생기면 서로 도와주며 소통하는 시간이 중요하다고 느꼈습니다.
이슈 정리를 깃헙으로 하며 서로의 상황을 알수 있어 좋았습니다

## PROBLEM
추가기능을 많이 넣지 못했다는 것이 많이 아쉬웠습니다. 제가 잠을 줄였다면 가능했을텐데, 피로가 누적이 돼서 그러지 못했습니다.
conflict 해결을 확실하게 하지 못하고 다른 팀원의 작업물을 지운 상태로 머지를 해버렸습니다.
팀원들이 사는 곳이 멀어서 다같이 얼굴을 대면해보지 못했다는 것이 많이 아쉬웠습니다. 다같이 모여서 프로젝트를 진행했다면, 학습적인 부분의 시너지가 더 많이 났을 것 같습니다.

너무 제 페이지에 대한 완성을 우선으로 생각하여 공통으로 해결해야 할 문제들을 너무 팀원분들께 부담드렸던 것 같습니다. 어떻게 보면 모두 같은 작업량을 가지고 프로젝트를 진행했지만 저만 크게 도와드린 게 없던 것 같아 죄송한 부분이 있었습니다. 이로 인해 다른 팀원분들이 늦은 시간까지 작업을 할 떄도 있었던 것 같습니다.

시간이 일주일만 더 있었으면 더 많은 기능을 구현하고 완벽한 결과물을 만들 수 있었는데 아쉽습니다.
제가 맡은 부분을 너무 완벽하게 하느라 작업 시간이 늦어져 다른 팀원 분들을 많이 도와 드리지 못한게 아쉽습니다.
범용성 있는 컴포넌트를 만들지 못해 아쉽습니다.

명세서에 맞게 완벽하게 구현을 못한거 같아 아쉽습니다.
프로젝트 명세에 맞춰 작업하느라 숙지가 필요한 부분을 제대로 짚고 가지 못하느 느낌이 들어 마음에 여유가 없었던거 같습니다.
마감 기한에 쫓겨 컨벤션 적용이 미흡한거 같습니다
전반적인 작업을 마치고 틀을 만들어 놓은 다음에 리팩토링 하며 작업하는 방법이 작업 능률에 있어 좋다고 느꼈습니다

## TRY
피로가 누적되지 않게 적절한 휴식을 취할 것입니다.
conflict는 꼼꼼하게 확인하고, 머지를 하기전에도 다시 한 번 더 꼼꼼하게 확인을 하는 습관을 들일 것입니다.

우선 제 페이지가 완성이 되면 빠르게 공통으로 작업할 수 있는 부분을 확인하고 먼저 해결하려고 하겠습니다. 해결하지 못할 수 있어서 도전을 못했지만, 다음부터는 사소한 부분이라도 먼저 나서서 해결하려고 하겠습니다.

컴포넌트를 만들 때 여러가지 상황을 고려해 범용성 있게 만들겠습니다.
생각하고 코드를 작성하는 과정을 단축시키기 위해 더 공부를 열심히 하겠습니다.

원활한 작업을 위해 어떻게 해야할지 고민해보고 나중에 리팩토링한다고 너무 밀어두지 말고 작업하는 순간 최대한 공부하고 적용해보려고 노력하겠습니다

**Comment by soyxxn** [comment]
### 3팀

### KEEP

- **열정적인 개별 기여**: 각자 맡은 부분에 대해 열정적으로 고민하고 좋은 방향으로 이끌어가는 팀원들의 노력.
- **고모군의 코드 리뷰**: 고모군님이 진행하는 성실한 코드 리뷰가 팀에 긍정적인 영향을 미치고 있음.
- **다양한 의견 교환**: 팀 내에서 활발하게 다양한 의견이 오가며, 이것이 긍정적인 분위기 조성에 기여하고 있음.
- **재미있고 긍정적인 분위기**: 팀원들 간의 유쾌하고 긍정적인 분위기 유지.
- **상호 배려**: 서로를 배려하는 팀 문화가 잘 형성되어 있음.
- **이모양의 꼼꼼함**: 이모양님이 꼼꼼하게 사항들을 체크하고 관리해주고 있는 점이 팀 작업의 퀄리티를 높이고 있음. 

### PROBLEM

- **전체 영향 우려**: 전체에 영향을 미치는 중요한 부분들(코드나 문서)에 대해 팀원들이 두려워하는 경향이 있음.
- **개별 고민의 결과**: 개인적인 고민을 너무 깊게 하여 나중에 그 결과를 후회하는 경우가 있음.
- **중복 작업**: 사전 협의 없이 중복되는 부분을 개발함으로써 발생하는 비효율성.
- **소통 부족**: 어려운 점이나 힘든 부분을 사전에 공유하지 않아 협력에 장애가 발생했음.
- **PR 과정의 지연**: PR 확인 및 머지 과정에서 소통이 지연되어 효율성 저하.

### TRY

- **직설적 의사소통**: 팀원들이 더 직설적으로 의사소통할 수 있는 분위기 조성을 시도.
- **공통 컴포넌트 개선**: 공통 컴포넌트의 효율성과 효과성을 높이는 작업에 집중.
- **체계적인 코드 리뷰**: 코드 리뷰 과정을 더 체계적이고 폼나게 진행하여 전반적인 코드 퀄리티 향상.
- **PR 과정 개선**: PR이 확인되면 빠르게 피드백을 공유하도록 하여 작업 흐름 개선.
- **우선순위 설정**: 프로젝트의 우선순위를 명확히 해 팀원들이 무엇을 먼저 해야 할지 인지할 수 있도록 할 것.

**Comment by GihoKo** [comment]
# 8팀

# **👍🏻**Keep

- 날이 지날 수록 체계적이고 잘 정리가 되는 것 같습니다.
- 서로 서로 잘 도와줍니다.
- 시간이 걸리더라도 포기하지 않고 어떻게든 만들어내려고 노력합니다.
- 다들 끈기있고 차분하게 해 낸 것 같습니다.
- 다들 엄청난 러닝커브에도 불구하고 누군가는 가르쳐주고 누군가는 열심히 배우려 노력하는 모습이 감동적이였습니다.
- 다들 어려운 프로젝트 속에서 새로운 기술까지 이것저것 공부하고 시도하시느라 고생많으셨습니다. 모두 잘 적용되진 못했지만 이번의 경험이 이후 성장해 나가는데 좋은 밑거름이 됐으면 좋겠습니다! 

우리팀의 가장 성실보스 팀장 기호님! 바지사장으로 스스로를 칭하셨지만 누구보다 팀을 잘 챙겨주셨습니다. 개발도 힘들고 바쁜가운데 어려운 역할인데도 성실하게 끝까지 이끌어주셔서 감사합니다. 다음에도 팀장하시는걸 추천합니다 ㅎㅎ 

우리팀 교수님, 모르는게 없는 형민님! 항상 앞선 지식으로 모르는거 많이 알려주시고 도입할 수 있게 도움주셔서 실력 향상에 큰 도움을 주셨습니다. 감사합니다! 다른 팀 가셔서도 지금처럼 많이 알려주고 공유해주시면 그 팀은 참  좋을 것 같아요

조용하지만 강력한 한방이 있는 은진님! 쉬울 줄 알았건만 어려움 한 가득이었던 파트, 힘들었을텐데 끝까지 해내셨습니다. 중간중간에 번뜩이는 아이디어로 문제해결에 도움을 주셔서 무사히 마쳤습니다. 감사합니다! 다른 팀 가서도 계속 좋은 아이디어 공유해주세요~!

우리팀의 숨은 재능러 민교님! 재밌는 드립과 웃음으로 팀에 좋은 분위기를 내는데 큰 도움이 됐습니다. 마지막 발표자료까지 이 여정을 마무리하는 화룡점정을 맡아주셨네요~ 감사했습니다!(발표때 함께하지  못해서 죄송했어요!) 다른 팀 발표도 민교님 발표면 긴장해야겠어요 ㅎㅎ


## **🤔**Problem

- 회의 시간이 너무 길었던 것 같습니다.
- 문제상황에 대한 적극적인 공유가 늦었던 것 같습니다.
- 밤샘은 이제 그만..
- 모두 체력관리를 하는데 있어서 많이 힘들었던 것 같습니다
- 첫 만남에 다들 잘하고 성실하고 모난 사람 없어보여서 이번 팀플은 무조건 잘될 것만 같았지만 생각보다 어려움도 많았고, 기대만큼의 결과물을 못뽑아낸게 못내 아쉽습니다. 저는 중간중간 체력도 멘탈도 잘 관리 안돼서 혼자 방황한 날도 있고 그랬네요 ㅎㅎ 그랬던 하루 이틀이 마지막 마무리에 부족했던 하루 이틀의 시간이 된 것 같아 속상하고 미안한 마음이 듭니다. 모두 고생하셨습니다!

## **🔥**Try

- 새로운 것들을 너무 많이 도입할 때 충분한 고민을 해야겠습니다.
- 적극적이고 즉각적인 공유를 통해 누군가 이미 겪었을 지도, 혹은 나중에 겪을 지도 모르는 문제를 쉽게 해결하도록 했으면 좋겠습니다.
- 예상되는 러닝커브를 좀 더 보수적으로 생각해 프로젝트를 해나가야할 것 같습니다.
- 이제 여유로우니 운동 꼭 시작하기! 밥 잘먹기!
- 저를 포함해서 모두 처음은 자신을 드러내는데 어려움이 큰 것 같습니다. 그것은 눈치, 탐색, 배려, 성격 등의 다양한 이름으로 나타나지만 결국 그 기간이 길어질 수록 협업의 효율이 나타나는데 오래걸린다는 생각이 이번에  들었습니다. 짧게나마 제가 겪은 바로는 여러분들은 모두 잘하고 알면 알수록 좋은 사람들이었습니다 ㅎㅎ 자신을 드러내는데 꺼리기보다는 먼저 드러내고 다가가는 용기를 갖는다면 더 좋은 결과에 기여하는 일원이 되지 않을까 하는 생각이 들었습니다. (이 말은 제 스스로에게 하는 말 이기도 합니다!)
